src/nanoemoji/write_variable_font.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Assembles UFOs into a variable font.
+
+Based on https://gist.github.com/anthrotype/c11515065a4aa6549f9d5f2dfdcf8f23"""
+
+from absl import app
+from fontTools import designspaceLib
+from nanoemoji import config
+import ufo2ft
+import ufoLib2
+
+
+def main(argv):
+    ufos = tuple(a for a in argv[1:] if a.endswith(".ufo"))
+
+    font_config = config.load()
+
+    designspace = designspaceLib.DesignSpaceDocument()
+
+    import pprint
+
+    pp = pprint.PrettyPrinter()
+
+    # define axes names, tags and min/default/max
+    axis_defs = [
+        dict(
+            tag=a.axisTag,
+            name=a.name,
+            minimum=min(
+                p.position
+                for m in font_config.masters
+                for p in m.position
+                if p.axisTag == a.axisTag
+            ),
+            default=a.default,
+            maximum=max(
+                p.position
+                for m in font_config.masters
+                for p in m.position
+                if p.axisTag == a.axisTag
+            ),
+        )
+        for a in font_config.axes
+    ]
+    pp.pprint(axis_defs)
+    for axis_def in axis_defs:
+        designspace.addAxisDescriptor(**axis_def)
+
+    axis_names = {a.axisTag: a.name for a in font_config.axes}
```

Comment:
in HOI, axisTags may not be unique. Not sure if how future-proof we want to be here.

Reply:
I fear we'd need much more extensive changes for this tool to support HOI so I'm OK with this not working for now. The sources don't really support it currently.